packages/core/src/script-executer.ts

```diff
@@ -0,0 +1,26 @@
+import path from 'path'
+import { randomBytes } from 'crypto'
+import { CommandExecuter, ExecResult, mkdir } from './command-executer'
+import { Logger } from './log'
+import { upload } from './upload-files'
+
+export type ScriptExecuter = (
+  script: string,
+  opts?: { env?: Record<string, string | undefined> },
+) => Promise<ExecResult>
+
+export const scriptExecuter = ({ exec, log }: {
+  exec: CommandExecuter
+  log: Logger
+}): ScriptExecuter => async (script, opts = {}) => {
+  const scriptFile = path.basename(script)
+  const destination = `/tmp/scripts/${scriptFile}.${randomBytes(16).toString('hex')}`
```

Comment:
> Use `os.tmpdir()` and `path.join`

this is a remote fs while these functions represent the local fs. it will break in windows. also, we can make assumptions about the remote machine (e.g, that it has /tmp).